src/UnwrapReadableStream.php

```diff
@@ -126,7 +126,7 @@ public function close()
         $this->closed = true;
 
         // try to cancel promise once the stream closes
-        if ($this->promise instanceof CancellablePromiseInterface) {
+        if ($this->promise instanceof CancellablePromiseInterface  || ($this->promise !== null && method_exists($this->promise, 'cancel') && $this->promise instanceof PromiseInterface)) {
```

Comment:
Just looked into different ways to add a proper type check that works across all supported Promise versions and came up with this suggestion in https://github.com/reactphp/promise-timer/pull/55. WDYT?

```suggestion
        if ($this->promise !== null && \method_exists($this->promise, 'cancel')) {
```

This allows us to remove all references to `CancellablePromiseInterface` (see also same change below and `use` at top).

Reply:
@clue Making those changes now :+1: